linkerd/proxy/tcp/src/forward.rs

```diff
@@ -27,23 +27,27 @@ impl<C> Forward<C> {
 impl<C, T, I> Service<(T, I)> for Forward<C>
 where
     C: Service<T>,
+    C::Response: Send + 'static,
+    C::Future: Send + 'static,
     C::Error: Into<Error>,
     C::Response: AsyncRead + AsyncWrite,
-    I: AsyncRead + AsyncWrite,
+    I: AsyncRead + AsyncWrite + Send + 'static,
 {
-    type Response = ();
+    type Response = Box<dyn Future<Item = (), Error = Error> + Send + 'static>;
```

Comment:
This is always the same type, so I don't think we should have to box it:
```suggestion
    type Response = ForwardFuture<I, C::Future>;
```